### PR TITLE
Clean up redundant PHP FPM pool configurations

### PR DESCRIPTION
**Pull request by WyriHaximus**
Since we're generating a configuration from a template file there is no
need to keep upstream PHP FPM pool configuration files lying around
that could case issues down the line.

# Usabilla PHP Docker Template

Reviewers: @usabilla/oss-docker

## Type

Please specify the type of changes being proposed:

| Q                 | A
| ----------------- | -----
| Documentation?    | no
| Dockerfile change?| yes
| Build feature?    | no
| Apply CVE Patch?  | no
| Remove CVE Patch? | no

### CODE REVIEW

File: Dockerfile-fpm [hunk 1]
```diff
@@ -1,5 +1,7 @@
 FROM php:7.2-fpm-alpine3.7 as fpm
 
+RUN rm /usr/local/etc/php-fpm.d/*.conf
```
Comment (renatomefi) [inline Dockerfile-fpm]: What's the before and after files?
As there's the docker.conf, zz-docker.conf I'm not sure if this is a too safe of a change?

Comment (bendavies) [reply Dockerfile-fpm]: they are copied in on live 48.

Comment (bendavies) [reply Dockerfile-fpm]: putting this `rm` on line 47 would be better, for context.

**Comment by renatomefi** [comment]
I've done some digging into this, there's actually one file docker.conf coming from upstream which has useful stuff, I wrote a test for it:
```python
@pytest.mark.php_fpm_exec
def test_php_fpm_default_config(host):
    config_effective = host.run("php-fpm -tt").stderr

    assert "access.log = /proc/self/fd/2" in config_effective
    assert "error_log = /proc/self/fd/2" in config_effective
    assert "log_limit = 8192" in config_effective
    assert "clear_env = no" in config_effective
    assert "catch_workers_output = yes" in config_effective
    assert "decorate_workers_output = no" in config_effective

```

With your changes it fails:
```bash
# php-docker-template git:(undefined/wyrihaximus/clean-up-fpm-pool-configurations) ✗ ./test-fpm.sh usabillabv/php:7.4-fpm-alpine3.11
Starting a container for 'usabillabv/php:7.4-fpm-alpine3.11'
================================================================================================== test session starts ==================================================================================================
platform linux -- Python 3.7.5, pytest-5.2.1, py-1.8.0, pluggy-0.13.0 -- /usr/bin/python3.7
rootdir: /tests, inifile: pytest.ini
plugins: testinfra-2.1.0
collected 39 items / 18 deselected / 21 selected                                                                                                                                                                        

container/test_user.py::test_userPresent[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                                                        [  4%]
container/php/test_config.py::test_php_ini_dir_environment_variable_is_set[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                      [  9%]
container/php/test_config.py::test_php_config_files_are_present[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                                 [ 14%]
container/php/test_config.py::test_php_base_config_is_present[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                                   [ 19%]
container/php/test_config.py::test_production_config_link[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                                       [ 23%]
container/php/test_config.py::test_config_files_are_loaded[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                                      [ 28%]
container/php/test_config.py::test_production_config_is_effective[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                               [ 33%]
container/php/test_config.py::test_fpm_configuration_is_effective[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                               [ 38%]
container/php/test_dev_config.py::test_configuration_is_not_present[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                             [ 42%]
container/php/test_dev_config.py::test_configuration_files_are_not_loaded[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                       [ 47%]
container/php/test_dev_config.py::test_configuration_is_not_effective[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                           [ 52%]
container/php/test_dev_config.py::test_xdebug_is_not_loaded[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                                     [ 57%]
container/php/test_fpm_entrypoint.py::test_php_fpm_exec_has_dumb_init[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                           [ 61%]
container/php/test_fpm_entrypoint.py::test_php_fpm_default_config[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] FAILED                                                               [ 66%]
container/php/test_fpm_entrypoint.py::test_php_fpm_templates_config_file[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                        [ 71%]
container/php/test_fpm_entrypoint.py::test_php_fpm_templates_config_file_all_vars[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                               [ 76%]
container/php/test_helper_scripts.py::test_php_images_contain_helper_scripts[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                    [ 80%]
container/php/test_helper_scripts.py::test_php_source_tarball_script[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                            [ 85%]
container/php/test_helper_scripts.py::test_php_fpm_status_is_enabled[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                            [ 90%]
container/php/test_helper_scripts.py::test_php_extension_script_for_rdkafka[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                     [ 95%]
container/php/test_helper_scripts.py::test_php_extension_script_for_pdo_pgsql[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] PASSED                                                   [100%]

======================================================================================================= FAILURES ========================================================================================================
________________________________________________________ test_php_fpm_default_config[docker://14401e5bb1096df98d068af7c35751a60be5884c8460f0c42a397d9d1fb91edc] _________________________________________________________

host = <testinfra.host.Host object at 0x7f6fd29deb50>

    @pytest.mark.php_fpm_exec
    def test_php_fpm_default_config(host):
        config_effective = host.run("php-fpm -tt").stderr
    
>       assert "access.log = /proc/self/fd/2" in config_effective
E       AssertionError: assert 'access.log = /proc/self/fd/2' in '[07-Apr-2020 14:51:48] NOTICE: [global]\n[07-Apr-2020 14:51:48] NOTICE: \tpid = undefined\n[07-Apr-2020 14:51:48] NOT...51:48] NOTICE:  \n[07-Apr-2020 14:51:48] NOTICE: configuration file /usr/local/etc/php-fpm.conf test is successful\n\n'

container/php/test_fpm_entrypoint.py:19: AssertionError
---------------------------------------------------------------------- generated xml file: /results/php-fpm-usabillabv/php:7.4-fpm-alpine3.11.xml -----------------------------------------------------------------------
===================================================================================== 1 failed, 20 passed, 18 deselected in 42.62s ======================================================================================
```

I'm closing this one, I'll see if I pick up a regression test plus a more narrowed clean!

**Pull request closed without merging**